File: airesources/JavaScript/RandomBot.js
// Basic bot
'use strict'
const fs = require('fs');
const hlt = require('./hlt');
const networking = require('./networking');
var myID = setTimeout(networking.getInit,200)[0],
	a = networking.sendInit("NodeJS Random Bot");

while (true) {
	var moves = [];
	networking.getFrame().then(function (gameMap) {
		for (var y = 0; y < gameMap.height; y++) {
			for (var x = 0; x < gameMap.width; x++) {
				if (gameMap.getSite(new hlt.Location(x,y)).owner == myID) {
					moves.append(new hlt.Move(new hlt.Location(x,y), (Math.floor(Math.random()*5))))
				}
			}
		}
		if (moves.len < 1) {
			console.log("NO MOVES")
		} else {
			networking.sendFrame(moves)
		}
	});


}

File: airesources/Kotlin/RandomBot.kt
import java.util.*

fun main(args : Array<String>) {
  RandomBot().start()
}

class RandomBot {

  fun start() {
    val network = Networking()
    val iPackage = network.getInit()
    val myID = iPackage.myID
    var gameMap = iPackage.map

    network.sendInit("RandomKotlinBot")

    while (true) {
      val moves = ArrayList<Move>()

      gameMap = network.getFrame()

      for (y in 0..gameMap.height - 1) {
        for (x in 0..gameMap.width - 1) {
          val site = gameMap.getSite(Location(x, y))
          if (site.owner == myID) {
            val dir = Direction.random()
            moves.add(Move(Location(x, y), dir))
          }
        }
      }
      network.sendFrame(moves)
    }
  }
}